tests/python_keep_opening_on_previous_line/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,243 @@
+    long_func_2_aaaaaaaaaaaaaaaaa(long_func_3_aaaaaaaaaaaaaa(1))
+)
+
+
+def foo():
+    return [
+        long_item_1_aaaaaaaaaaaa,
+        long_item_1_aaaaaaaaaaaa,
+        long_item_1_aaaaaaaaaaaa
+    ]
+
+
+def foo():
+    yield [
+        long_item_1_aaaaaaaaaaaa,
+        long_item_1_aaaaaaaaaaaa,
+        long_item_1_aaaaaaaaaaaa
+    ]
+
+`;
+
+exports[`keep_opening_on_previous_line.py 2`] = `
+my_long_variable_aaaaaaaaaaaaaaaaaaaaaa = my_long_value_aaaaaaaaaaaaaaaaaaaaaaaaaa
+
+my_long_variable_aaaaaaaaaaaaaaaaaaaaaa = 1 / my_long_value_aaaaaaaaaaaaaaaaaaaaaaaaaa
+
+my_long_variable_1_aaaaaaaaaaaaaaaaaaaaaa = my_long_variable_2_aaaaaaaaaaaaaaaaaaaaaa = my_long_value_aaaaaaaaaaaaaaaaaaaaaaaaaa
+
+my_long_variable_1_aaaaaaaaaaaaaaaaaaaaaa = my_long_variable_2_aaaaaaaaaaaaaaaaaaaaaa = my_long_variable_3_aaaaaaaaaaaaaaaaaaaaaa = my_long_value_aaaaaaaaaaaaaaaaaaaaaaaaaa
+
+a = [long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa]
+
+a = b = [long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa]
+
+my_long_variable_1_aaaaaaaaaaaaaaaaaaaaaa = my_long_variable_2_aaaaaaaaaaaaaaaaaaaaaa = [long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa]
+
+foo(bar(baz([long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa])))
+
+foo(bar(baz([long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa], 1)))
+
+foo(bar(baz([long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa]), 1))
+
+foo(bar(baz([long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa])), 1)
+
+long_func_1_aaaaaaaaaaaaa(long_func_2_aaaaaaaaaaaaaaaaa(long_func_3_aaaaaaaaaaaaaa(1)))
+
+def foo():
+    return [long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa]
+
+
+def foo():
+    yield [long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa, long_item_1_aaaaaaaaaaaa]
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+my_long_variable_aaaaaaaaaaaaaaaaaaaaaa = \\
+    my_long_value_aaaaaaaaaaaaaaaaaaaaaaaaaa
+
+my_long_variable_aaaaaaaaaaaaaaaaaaaaaa = \\
+    1 / my_long_value_aaaaaaaaaaaaaaaaaaaaaaaaaa
+
+my_long_variable_1_aaaaaaaaaaaaaaaaaaaaaa = \\
+    my_long_variable_2_aaaaaaaaaaaaaaaaaaaaaa = \\
+    my_long_value_aaaaaaaaaaaaaaaaaaaaaaaaaa
+
+my_long_variable_1_aaaaaaaaaaaaaaaaaaaaaa = \\
+    my_long_variable_2_aaaaaaaaaaaaaaaaaaaaaa = \\
+    my_long_variable_3_aaaaaaaaaaaaaaaaaaaaaa = \\
+    my_long_value_aaaaaaaaaaaaaaaaaaaaaaaaaa
+
+a = [
+    long_item_1_aaaaaaaaaaaa,
+    long_item_1_aaaaaaaaaaaa,
+    long_item_1_aaaaaaaaaaaa
+]
+
+a = b = [
+    long_item_1_aaaaaaaaaaaa,
+    long_item_1_aaaaaaaaaaaa,
+    long_item_1_aaaaaaaaaaaa
+]
+
+my_long_variable_1_aaaaaaaaaaaaaaaaaaaaaa = \\
+    my_long_variable_2_aaaaaaaaaaaaaaaaaaaaaa = [
+        long_item_1_aaaaaaaaaaaa,
+        long_item_1_aaaaaaaaaaaa,
+        long_item_1_aaaaaaaaaaaa
+    ]
+
+foo(bar(baz([
```

Comment:
The way prettier algorithm has been designed is to break the outermost group first, so this should look the following way:

```js
foo(
  bar(
    baz(
      ...
    )
  )
)
```

If you want it to print it the way you are displaying it right now, prettier is --not-- the right algorithm. And frankly, I don't know what the right algorithm is...

Reply:
Yeah that's a good point. How crucial is the inlineless of `foo(bar(baz(` to Python style?